### PR TITLE
BUGFIX: Fix salinity input co2 module

### PR DESCRIPTION
**Pull request by totto82**
This sets the salinity in the CO2Store module
It also make it possible to run CO2STORE with multiple pvt regions without crashing

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

File: opm/material/fluidsystems/blackoilpvt/Co2GasPvt.cpp [hunk 1]
```diff
@@ -60,8 +55,15 @@ initFromState(const EclipseState& eclState, const Schedule&)
...
 
-    gasReferenceDensity_[regionIdx] = CO2::gasDensity(T_ref, P_ref, extrapolate);
-    brineReferenceDensity_[regionIdx] = Brine::liquidDensity(T_ref, P_ref, salinity_[regionIdx], extrapolate);
+    std::size_t numRegions = eclState.runspec().tabdims().getNumPVTTables();
+    setNumRegions(numRegions);
+    for (size_t regionIdx = 0; regionIdx < numRegions; ++regionIdx) {
+        // Currently we only support constant salinity converted to mass fraction
+        salinity_[regionIdx] = eclState.getCo2StoreConfig().salinity();
+        gasReferenceDensity_[regionIdx] = CO2::gasDensity(T_ref, P_ref, extrapolate);
+        brineReferenceDensity_[regionIdx] = Brine::liquidDensity(T_ref, P_ref, salinity_[regionIdx], extrapolate);
```
Comment (svenn-t) [inline opm/material/fluidsystems/blackoilpvt/Co2GasPvt.cpp:64]: `brineReferenceDensity_` must be calculated with Ezrokhi's method when DENAQA is given in the deck

**Review by svenn-t — COMMENTED** [review]
To be consistent with the `BrineCo2Pvt` module, `brineReferenceDensity_` must be calculated using DENAQA if it is in the deck.

**Comment by totto82** [comment]
jenkins build this please

File: opm/material/fluidsystems/blackoilpvt/Co2GasPvt.hpp [hunk 2]
```diff
@@ -330,8 +330,30 @@ class Co2GasPvt
...
+        if (denaqa.empty())
+            return;
+
+        enableEzrokhiDensity_ = true;
+        ezrokhiDenNaClCoeff_ = {static_cast<Scalar>(denaqa[0].getC0("NACL")), 
+                                static_cast<Scalar>(denaqa[0].getC1("NACL")), 
+                                static_cast<Scalar>(denaqa[0].getC2("NACL"))};
+        ezrokhiDenCo2Coeff_ = {static_cast<Scalar>(denaqa[0].getC0("CO2")), 
+                               static_cast<Scalar>(denaqa[0].getC1("CO2")), 
+                               static_cast<Scalar>(denaqa[0].getC2("CO2"))};
```
Comment (svenn-t) [inline opm/material/fluidsystems/blackoilpvt/Co2GasPvt.hpp:344]: `ezrokhiDenCo2Coeff_` not used in `Co2GasPvt` so can be removed

File: opm/material/fluidsystems/blackoilpvt/Co2GasPvt.cpp [hunk 3]
```diff
@@ -47,21 +47,32 @@ initFromState(const EclipseState& eclState, const Schedule&)
...
 
     // Throw an error if STCOND is not (T, p) = (15.56 C, 1 atm) = (288.71 K, 1.01325e5 Pa)
     if (T_ref != Scalar(288.71) || P_ref != Scalar(1.01325e5)) {
         OPM_THROW(std::runtime_error, "CO2STORE/CO2SOL can only be used with default values for STCOND!");
     }
+    setEzrokhiDenCoeff(eclState.getCo2StoreConfig().getDenaqaTables());
+
+    std::size_t numRegions = eclState.runspec().tabdims().getNumPVTTables();
+    setNumRegions(numRegions);
+    for (size_t regionIdx = 0; regionIdx < numRegions; ++regionIdx) {
```
Comment (svenn-t) [inline opm/material/fluidsystems/blackoilpvt/Co2GasPvt.cpp:61]: `std::size_t`

File: opm/material/fluidsystems/blackoilpvt/BrineCo2Pvt.cpp [hunk 4]
```diff
@@ -100,23 +94,32 @@ initFromState(const EclipseState& eclState, const Schedule&)
...
-    if (enableEzrokhiDensity_) {
-        const Scalar& rho_pure = H2O::liquidDensity(T_ref, P_ref, extrapolate);
-        const Scalar& nacl_exponent = ezrokhiExponent_(T_ref, ezrokhiDenNaClCoeff_);
-        brineReferenceDensity_[regionIdx] = rho_pure * pow(10.0, nacl_exponent * salinity_[regionIdx]);
-    }
-    else {
-        brineReferenceDensity_[regionIdx] = Brine::liquidDensity(T_ref, P_ref, salinity_[regionIdx], extrapolate);
+    std::size_t numRegions = eclState.runspec().tabdims().getNumPVTTables();
+    setNumRegions(numRegions);
+    for (size_t regionIdx = 0; regionIdx < numRegions; ++regionIdx) {
```
Comment (svenn-t) [inline opm/material/fluidsystems/blackoilpvt/BrineCo2Pvt.cpp:99]: `std::size_t`

**Review by svenn-t — COMMENTED** [review]
Just have a few more minor comments, then it looks good to me.

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
jenkins build this update_data please

**Comment by totto82** [comment]
jenkins built this opm-tests=1180 please

**Comment by bska** [comment]
jenkins build this opm-tests=1180 please

**Pull request merged**